pkg/agent/controller/agent.go

```diff
@@ -564,3 +579,32 @@ func (a *Controller) filterLocalEndpointSlices(obj runtime.Object, numRequeues i
 
 	return obj, false
 }
+
+func (a *Controller) getGlobalIP(service *corev1.Service) (ip, reason, msg string) {
+	if a.globalnetEnabled {
+		ip = a.getGlobalIPFromService(service)
```

Comment:
Cool. So we are maintaining backward compatibility.

Reply:
Personally, I'd prefer to remove this altogether when we switch. Only reason I left it in for now is to not break things till we switch to v2. Since we don't want to support v1 at all, it is better to avoid problems that can crop up from trying to half heartedly support v1 + v2.